### ISSUE TITLE
[HxSearchBox] Unable to implement text query search with suggestions

### ISSUE THREAD

**Issue by alexax578** [havit/Havit.Blazor#575]
I want to use HxSearchBox to implement a search component with the following behavior:

- Allow user to search by text query
- Allow user to select an item from the suggestions list
- Always retain the inputted value

I am unable to do so because upon selecting an item from the suggestions I handle ItemSelected event callback by setting the value of the item to the text query (to retain the value in the input) but then ClearInputAsync is called inside the component which in return triggers TextQueryChanged that I use to intercept searches by text query.

What I propose is to remove ClearInputAsync from the HandleItemSelected method on BtSearchBox or add a parameter which allows configuration of this behavior.

**Comment by crdo** [comment]
@alexax578 is it not what have just been implemented to the HxAutosuggest in the last release?

**Comment by alexax578** [comment]
> @alexax578 is it not what have just been implemented to the HxAutosuggest in the last release?

I don't think so. HxAutosuggest does not allow submitting the value without selecting an item (when you press enter it automatically selects the first available item from the suggestions instead of just going with the input the user typed).

**Comment by hakenr** [comment]
In the end, I decided to remove the `ClearInputAsync()` from `HandleItemSelected()` and replace it with setting the input-text (`TextQuery`) with `Title` of the selected item. The final decision was influenced by the fact that the `TextQuery`-mode leaves the input-text populated and this was a step to unify the behavior.

**Note:** This is a breaking UX change and those projects expecting the `HxSearchBox` to clear after item selection have to set the `TextQuery` to String.Empty on their own (will test in GOR-G3 and consider adding the `bool ClearTextQueryAfterItemSelection` parameter) to make those scenarios easier to implement.

(Will make some usability test with the pre-release.)

**Comment by hakenr** [comment]
@alexax578 give it a try: 4.1.2-pre1

**Comment by hakenr** [comment]
After testing the developer experience I decided to introduce new parameter:
```csharp
/// <summary>
/// Behavior when the item is selected.
/// </summary>
[Parameter] public SearchBoxItemSelectionBehavior? ItemSelectionBehavior { get; set; }
```

with values
```csharp
public enum SearchBoxItemSelectionBehavior
{
	SelectAndClearTextQuery = 0,
	SelectAndReplaceTextQueryWithItemTitle = 1,
}
```

The default behavior will remain `SelectAndClearTextQuery` to avoid the UX breaking change.

Will be published in 4.1.2-pre2.
Suggestions for better parameter/values naming are welcome. @jirikanda @crdo @alexax578

**Comment by alexax578** [comment]
@hakenr great! This seems to resolve my issue.
